packages/netlify-cms-core/src/components/Collection/Entries/EntryListing.js

```diff
@@ -68,15 +69,15 @@ export default class EntryListing extends React.Component {
   };
 
   render() {
-    const { collections } = this.props;
+    const { collections, page } = this.props;
 
     return (
       <div>
         <CardsGrid>
           {Map.isMap(collections)
             ? this.renderCardsForSingleCollection()
             : this.renderCardsForMultipleCollections()}
-          {this.hasMore() && <Waypoint onEnter={this.handleLoadMore} />}
+          {this.hasMore() && <Waypoint key={page} onEnter={this.handleLoadMore} />}
```

Comment:
This is fix number 2 - remount `Waypoint` to force `handleLoadMore` on page changes (required so we keep fetching pages until we have some entries).